src/main/webapp/js/instructorStudentListAjax.js

```diff
@@ -70,14 +74,39 @@ var seeMoreRequest = function(e) {
                         displayIcon.html('<img height="25" width="25" src="/images/ajax-preload.gif">')
                     },
                     error: function() {
-                        numStudents -= courseNumStudents;
+                        // Handle error and allow retry
                     },
                     success: function(data) {
                         $(panelBody[0]).html(data);
+
+                        // Count number of students retrieved
+                        courseNumStudents = numStudentsRetrieved();
+                        $('#numStudents-' + courseIdx).val(courseNumStudents);
+
+                        // If number of students shown is already more than the limit
+                        // Do not show more, even if we can retrieve it.
+                        if (numStudents >= STUDENT_LIMIT) {
+                            // Failed to load
+                            courseCheck.prop('checked', false);
+                            setStatusMessage(PERFORMANCE_ISSUE_MESSAGE, true);
+                            displayIcon.html('');
```

Comment:
Seems to be a repeat of code in lines 119-122. Might be good to extract out a function for this, or write a comment, so that it's more obvious what this is for